### PR TITLE
Remove 1.23 dashboards and job

### PR DESCRIPTION
**Pull request by ramrodo**
Since v1.23 Kubernetes is out of support now, we should remove the 1.23 dashboards from testgrid and its Job

This PR is the following one of https://github.com/kubernetes/test-infra/pull/29172

closes https://github.com/kubernetes/test-infra/issues/29242

/sig release
/area release-eng

### CODE REVIEW

**Comment by ramrodo** [comment]
cc: @cici37 @gracenng @kubernetes/release-engineering

**Comment by cici37** [comment]
Other release jobs and testgrid settings should not be touched if the goal is to remove 1.23..

**Comment by ramrodo** [comment]
> Other release jobs and testgrid settings should not be touched if the goal is to remove 1.23..

@cici37, we should update the []() since the stable4 should be 1.24 now (instead of 1.23). Otherwise, [there is an error at Prow](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29230/pull-test-infra-unit-test/1645537532269563904#1:build-log.txt%3A565):

```
could not apply prowjob annotations:
couldn't find dashboard "sig-release-1.23-blocking" for job "ci-kubernetes-e2e-gce-cos-k8sstable4-alphafeatures"
FAIL	k8s.io/test-infra/config/tests/testgrids	2.881s 
```

**Comment by BenTheElder** [comment]
We should be updating the generated file, especially as the input spec changed earlier in this PR and merely running the generator would produce different results.

1.23 should be dropped from all of these jobs.

File: releng/test_config.yaml [hunk 1]
```diff
@@ -251,30 +251,30 @@ k8sVersions:
     - --extract-ci-bucket=k8s-release-dev
     version: master
   beta:
+    args:
+    - --extract=ci/latest-1.28
```
Comment (BenTheElder) [inline releng/test_config.yaml:255]: hmmm I don't think we should have 1.28 yet?

**Comment by cici37** [comment]
Seems like the auto generated yaml still binds to the wrong version: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/generated/generated.yaml
[generate_tests.py](https://github.com/kubernetes/test-infra/blob/master/releng/generate_tests.py) seems the one to make it right. 

And also, we should not have 1.28 as Ben mentioned above. I forgot how we use beta before, seems like we were using it for the current released version til the next branch is cut in the past.

**Comment by BenTheElder** [comment]
Can we start with a more scoped PR to the other 1.23 jobs aside from the dashboard and the generated jobs?
And then track down who owns this job generator and get those resolved?

**Comment by cici37** [comment]
> Seems like the auto generated yaml still binds to the wrong version: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/generated/generated.yaml
> [generate_tests.py](https://github.com/kubernetes/test-infra/blob/master/releng/generate_tests.py) seems the one to make it right.

The test will not pass if we didn't get generated jobs right with 1.23 removal. I agree we should start with minimized pr to remove 1.23 but I guess the above suggestion is still needed.

File: releng/test_config.yaml (hunk 1 above)
Comment (xmudrii) [reply releng/test_config.yaml:255]: The marker already exists in fact (https://dl.k8s.io/ci/latest-1.28.txt) and points to the latest commit in k/k at the time of writing this comment.

**Comment by xmudrii** [comment]
> And also, we should not have 1.28 as Ben mentioned above. I forgot how we use beta before, seems like we were using it for the current released version til the next branch is cut in the past.

We discussed it on Slack a while ago in some thread and this is wrong. The released version should switch to stable1 as soon as .0 is out, but we were forgetting to do so in the past. That said, proposed changes are correct.

I'm wondering why is https://github.com/ramrodo/test-infra/commit/39402344f8646b1fc8b5b01ac83109cfa7865005 reverted? Was that file manually changed so it got reverted?

**Comment by cici37** [comment]
A separate question, I tried to run 
```
go run ./releng/config-forker   --job-config $(pwd)/config/jobs   --version 1.27   --go-version 1.20.2   --output $(pwd)/config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml
```
which gives me different changes than `make -C releng prepare-release-branch`. Which one is the source of truth? Are they going through different code path? @xmudrii

**Comment by xmudrii** [comment]
@cici37 config-forker is only half of the job, there's also config-rotator that takes care of existing jobs (IIRC). The make command unifies both forker and rotator, so that's the reason why results are different.

**Comment by ramrodo** [comment]
> I'm wondering why is [ramrodo@3940234](https://github.com/ramrodo/test-infra/commit/39402344f8646b1fc8b5b01ac83109cfa7865005) reverted? Was that file manually changed so it got reverted?

I reverted that commit since the script added the `1.28` reference. So, my initial thought was that something was wrong.

**Comment by xmudrii** [comment]
@ramrodo Can you try reverting that revert commit? I think that we'll get tests passing, from there we can figure out what to do with 1.28.

File: config/jobs/kubernetes/generated/generated.yaml [hunk 2]
```diff
@@ -224,7 +224,7 @@ periodics:
   cluster: k8s-infra-prow-build
   annotations:
     testgrid-tab-name: gce-cos-k8sbeta-slow
-    testgrid-dashboards: sig-release-1.27-informing
+    testgrid-dashboards: sig-release-1.28-informing
     testgrid-num-failures-to-alert: '6'
 - tags:
```
Comment (xmudrii) [inline config/jobs/kubernetes/generated/generated.yaml:229]: @ramrodo Can you try commenting all jobs above this one and then pushing that?

Comment (xmudrii) [reply config/jobs/kubernetes/generated/generated.yaml:229]: I'm aware this is autogenerated file, but let's give it a try and iterate from there.

Comment (ramrodo) [reply config/jobs/kubernetes/generated/generated.yaml:229]: Sure

Comment (ramrodo) [reply config/jobs/kubernetes/generated/generated.yaml:229]: I'm wondering if adding the `#` characters made this test fail at converting to JSON or it is validating the integrity of the file:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29230/pull-test-infra-prow-checkconfig/1651730089395621888

Comment (xmudrii) [reply config/jobs/kubernetes/generated/generated.yaml:229]: https://github.com/kubernetes/test-infra/pull/29230#discussion_r1179797510

File: config/jobs/kubernetes/generated/generated.yaml [hunk 3]
```diff
@@ -1,231 +1,231 @@
...
-      - --gcp-node-image=gci
-      - --extract=ci/latest-1.27
-      - --extract-ci-bucket=k8s-release-dev
-      - --timeout=120m
-      - --test_args=--ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
-      - --ginkgo-parallel=30
-      - --env=ENABLE_CACHE_MUTATION_DETECTOR=true
-      env:
-      image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230406-23cb1879e3-master
-      resources: &id001
```
[the remark this answers is not in the document]
Comment (ramrodo) [reply config/jobs/kubernetes/generated/generated.yaml:70]: @xmudrii I'll do it with `id002` also

**Comment by xmudrii** [comment]
I did a write up on this issue in #29387, please check it and comment there on opinions and next steps.

**Comment by ramrodo** [comment]
Closing in favor of https://github.com/kubernetes/test-infra/pull/29387

**Pull request closed without merging**